src/components/calcite-checkbox/calcite-checkbox.tsx

```diff
@@ -39,7 +39,7 @@ export class CalciteCheckbox {
 
   @Watch("checked")
   checkedWatcher(newChecked: boolean): void {
-    newChecked ? this.input.setAttribute("checked", "") : this.input.removeAttribute("checked");
+    this.input.checked = newChecked;
```

Comment:
☝️ the property is the real source of truth since the attribute is only really used for setting the initial value.

> A Boolean attribute indicating whether or not this checkbox is checked by default (when the page loads). It does not indicate whether this checkbox is currently checked: if the checkbox’s state is changed, this content attribute does not reflect the change. (Only the HTMLInputElement’s checked IDL attribute is updated.)
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox